### ISSUE TITLE
Bitwise atomics tests in Mellanox suite causing Travis failure

### ISSUE THREAD

**Issue by davidozog** [Sandia-OpenSHMEM/SOS#725]
~~With `gcc` (but strangely not `icc`),~~ the new bitwise atomics tests in [tests-mellanox](https://github.com/openshmem-org/tests-mellanox) print "NONE" and return error code 9, which results a failure in Travis.

For example:
```
...
PASS   atomic     longlong_inc   shmem_longlong_inc(): A simple test.
PASS   atomic     stress         Atomic stress test.
NONE   atomic     int_xor        shmem_int_xor(): A simple test.
NONE  atomic     long_xor       shmem_long_xor(): A simple test.
NONE  atomic     longlong_xor   shmem_longlong_xor(): A simple test.
NONE  atomic     int_fxor       shmem_int_fxor(): A simple test.
NONE  atomic     long_fxor      shmem_long_fxor(): A simple test.
NONE  atomic     longlong_fxor  shmem_longlong_fxor(): A simple test.
NONE  atomic     int_fand       shmem_int_fand(): A simple test.
NONE  atomic     long_fand      shmem_long_fand(): A simple test.
NONE  atomic     longlong_fand  shmem_longlong_fand(): A simple test.
NONE  atomic     int_and        shmem_int_and(): A simple test.
NONE  atomic     long_and       shmem_long_and(): A simple test.
NONE  atomic     longlong_and   shmem_longlong_and(): A simple test.
NONE  atomic     int_for        shmem_int_for(): A simple test.
NONE  atomic     long_for       shmem_long_for(): A simple test.
NONE  atomic     longlong_for   shmem_longlong_for(): A simple test.
NONE  atomic     int_or         shmem_int_or(): A simple test.
NONE  atomic     long_or        shmem_long_or(): A simple test.
NONE  atomic     longlong_or    shmem_longlong_or(): A simple test.

**********************************
* Passed: 22
* Failed: 18
* Skipped: 0
* Ignored: 0
* Incompat:0
* Total: 40
```

SOS should have the bitwise atomics available, so I'm not yet sure what's causing this.  Any ideas @hoopoepg, @miked-mellanox, or @jdinan?  Looks like these where merged today in this PR: https://github.com/openshmem-org/tests-mellanox/pull/33

**Comment by hoopoepg** [comment]
@davidozog it returns none in case if OpenMPI hasn't new atomics implementation...
I will update tests to silently ignore missing API

**Comment by hoopoepg** [comment]
@davidozog could you try this PR: https://github.com/openshmem-org/tests-mellanox/pull/34
it should work properly on OpenMPI v3.1-

**Comment by davidozog** [comment]
Thanks!I  believe we're using MPICH Hydra v3.2 in Travis.  I'm re-running now to see what happens.

**Comment by davidozog** [comment]
It appears that the latest PR ([#34](https://github.com/openshmem-org/tests-mellanox/pull/34)) successfully skips these tests.  However, it would be nice to run them as a part of our SOS tests if possible...

I believe the problem might be that the type names in these atomics do not correspond to the bitwise type names in the OpenSHMEM v1.4 Specification.  Specifically, the symbols you look for in `AC_CHECK_DECL` (in the configure.ac file) are:
```
shmem_int_atomic_*, shmem_long_atomic_*, shmem_longlong_atomic_*
```
But the type names for bitwise atomics in Table 6 of the specification suggest these should be:
```
shmem_uint_atomic_*, shmem_ulong_atomic_*, shmem_ulonglong_atomic_*, shmem_int32_atomic_*, shmem_int64_atomic_*, shmem_uint32_atomic_*, shmem_uint64_atomic_*
```

If I'm not mistaken, those should be changed to conform to the spec.

**Comment by davidozog** [comment]
Yes, this was fixed by tests-mellanox [PR (#35)](https://github.com/openshmem-org/tests-mellanox/pull/35/files).
